js/components/utils.js

```diff
@@ -6,6 +6,127 @@
+   * min if it falls below, max if it falls above.
+   */
+  constrainToRange: function(value, min, max) {
+    value = this.ensureNumber(value, 0);
+    min = this.ensureNumber(min, 0);
+    max = this.ensureNumber(max, 0);
+    return Math.min(Math.max(min, value), max);
+  },
+
+  /**
+   * Returns the currentTime and totalTime values in HH:MM format that can be used for
+   * a video time display UI or for ARIA labels.
+   * Note that the meaning of these values changes depending on the type of video:
+   * VOD
+   *  currentTime: Formatted value of current playhead
+   *  totalTime: Formatted value of video duration
+   * Live - No DVR
+   *  currentTime: null
+   *  totalTime: null
+   * Live - DVR - useNegativeDvrOffset === true
+   *  currentTime: Formatted value of the negative offset from the live playhead
+   *  totalTime: null
+   * Live - DVR - useNegativeDvrOffset === false
+   *  currentTime: Formatted value of the current playhead relative to max time shift
+   *  totalTime: Formatted value of the total duration of the DVR window
+   *
+   * @function getTimeDisplayValues
+   * @param {Number} currentPlayhead The current value of the playhead in seconds.
+   * @param {Number} duration The total duration of the video in seconds.
+   * @param {Boolean} isLiveStream Indicates whether the video is a livestream or not.
+   * @param {Number} useNegativeDvrOffset Whether to display DVR progress as a negative offset value or not.
+   * @return {Object} An object with currentTime and totalTime properties in HH:MM format. Either of these
+   * might be null depending on the conditions above.
+   */
+  getTimeDisplayValues: function(currentPlayhead, duration, isLiveStream, useNegativeDvrOffset) {
+    currentPlayhead = this.ensureNumber(currentPlayhead);
+    duration = this.ensureNumber(duration, 0);
+
+    var currentTime = null;
+    var totalTime = null;
+
+    var currentPlayheadInt = parseInt(currentPlayhead, 10);
+    var currentPlayheadTime = isFinite(currentPlayheadInt) ? Utils.formatSeconds(currentPlayheadInt) : null;
+    var timeShift = (currentPlayhead || 0) - duration;
+
+    if (duration) {
+      totalTime = Utils.formatSeconds(duration);
+    }
+
+    if (isLiveStream) {
+      // Checking timeShift < 1 second (not === 0) as processing of the click after we
+      // rewinded and then went live may take some time.
+      var isLiveNow = Math.abs(timeShift) < 1;
+
+      if (useNegativeDvrOffset) {
+        // We don't show current time unless there is a time shift when using
+        // negative DVR offset
+        currentTime = isLiveNow ? null : Utils.formatSeconds(timeShift);
+      } else {
+        // When not using negative DVR offset, DVR progress is shown in the usual
+        // "current time of total time" format, with total time set to the size of DVR window
+        currentTime = isLiveNow ? totalTime : Utils.formatSeconds(Math.ceil(duration + timeShift));
```

Comment:
Should we also check that total time is not null?

Reply:
Good catch

Reply:
Are you addressing this in a future PR?

Reply:
@aeng7 yes I will include this in the next one.